php/php.editor/tools/ASTPHP5Parser.cup

```diff
@@ -3022,7 +3045,15 @@ method_modifiers ::=
     RESULT = Integer.valueOf(result);
 :}
 
-| af_modifiers:fModifier ppp_modifiers:sModifier
+| abstract_modifier:aModifier ppp_modifiers:sModifier
+{:
+    int result = 0;
+    result |= aModifier.intValue();
+    result |= sModifier.intValue();
+    RESULT = Integer.valueOf(result);
+:}
+
+| final_modifier:fModifier ppp_modifiers:sModifier
```

Comment:
`method_modifiers ::=` part
Why did separate `abstract` and `final`?
The reason: because the following cases are marked as syntax errors...
(I'm not sure the cause...)
```php
final public function method() {}
final public static function method() {}
```